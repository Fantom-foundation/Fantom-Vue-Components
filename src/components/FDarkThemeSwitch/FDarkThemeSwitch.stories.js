import FDarkThemeSwitch from './FDarkThemeSwitch.vue';

export default {
    title: 'FDarkThemeSwitch',
    component: FDarkThemeSwitch,
};

export const Default = () => ({
    components: { FDarkThemeSwitch },
    template: `
        <div>
            <f-dark-theme-switch />
            <f-dark-theme-switch label="with label" />
        </div>
    `,
});

export const Size = () => ({
    components: { FDarkThemeSwitch },
    template: `
        <div>
            <f-dark-theme-switch option-size="large" />
            <f-dark-theme-switch />
            <f-dark-theme-switch option-size="small" />
            <f-dark-theme-switch option-size="mini" />
        </div>
    `,
});

export const Colors = () => ({
    components: { FDarkThemeSwitch },
    template: `
        <div>
            <f-dark-theme-switch :style="blueStyle" />
            <f-dark-theme-switch :style="purpleStyle" />
        </div>
    `,
    data() {
        return {
            blueStyle: {
                '--fdarkthemeswitch-color': '#08209e',
                '--fdarkthemeswitch-color-hsl': '230, 90%, 33%',
            },
            purpleStyle: {
                '--fdarkthemeswitch-color': '#800E99',
                '--fdarkthemeswitch-color-hsl': '289, 83%, 33%',
            },
        };
    },
});
