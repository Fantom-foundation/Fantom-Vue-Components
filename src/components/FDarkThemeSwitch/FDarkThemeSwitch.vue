<template>
    <span class="fdarkthemeswitch">
        <f-toggle-button v-bind="$attrs" v-on="$listeners" />
    </span>
</template>

<script>
import FToggleButton from '../FToggleButton/FToggleButton.vue';

/**
 * Toggle button with sun and moon icons
 */
export default {
    name: 'FDarkThemeSwitch',

    components: { FToggleButton },

    inheritAttrs: false,

    model: {
        prop: 'modelValue',
        event: 'change',
    },
};
</script>

<style lang="scss">
@use 'style';
</style>
