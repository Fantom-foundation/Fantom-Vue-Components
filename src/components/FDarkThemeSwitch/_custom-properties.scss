.fdarkthemeswitch {
    --fdarkthemeswitch-color: var(--f-color-primary-5, #1a6aff);
    --fdarkthemeswitch-color-hsl: var(--f-color-primary-h, 219), var(--f-color-primary-s, 100%), var(--f-color-primary-l, 55%);
    //--fdarkthemeswitch-color-hsl: var(--f-togglebtn-theme-color-hsl);
}
