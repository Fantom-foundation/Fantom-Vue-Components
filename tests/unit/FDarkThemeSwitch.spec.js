import { mount } from '@vue/test-utils';
import FDarkThemeSwitch from '@/components/FDarkThemeSwitch/FDarkThemeSwitch.vue';

let wrapper = null;

function destroyWrapper() {
    if (wrapper) {
        wrapper.destroy();
    }
}

function createWrapper({ propsData = {} } = {}) {
    destroyWrapper();

    wrapper = mount(FDarkThemeSwitch, {
        propsData,
    });
}

describe('FDarkThemeSwitch', () => {
    it('should render correctly', () => {
        createWrapper({ propsData: { id: 'foo' } });

        expect(wrapper.element).toMatchSnapshot();
    });
});
