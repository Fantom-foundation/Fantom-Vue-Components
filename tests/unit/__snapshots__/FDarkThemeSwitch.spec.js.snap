// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FDarkThemeSwitch should render correctly 1`] = `
<span
  class="fdarkthemeswitch"
>
  <label
    class="ftogglebutton togglebtn togglebtn-v1"
  >
    <input
      class="togglebtn_input"
      id="foo"
      name=""
      type="checkbox"
      value=""
    />
     
    <span
      class="togglebtn_btn"
    />
     
    <span
      class="togglebtn_label"
    >
      
    </span>
  </label>
</span>
`;
